### ISSUE TITLE
Parsing of first header line fails if using UTF-8 with BOM

### ISSUE THREAD

**Issue by k--kato** [uniVocity/univocity-parsers#154]
version: 2.4.1

When using an UTF-8 with BOM parsing of the first header line seems to fail.

**Comment by jbax** [comment]
Thanks for reporting the issue. I saw this some time ago but had forgotten about it.

You can use the following workaround to skip the BOM marker:

```
    InputStream in = new FileInputStream("/path/to/file.csv", "UTF-8");

    //skip UTF-8 BOM bytes:
    if(in.read() == 239 & in.read() == 187 & in.read() == 191){
         System.out.println("UTF-8 with BOM, bytes discarded");
    }
```

Will be fixed on version 2.5.0

**Comment by k--kato** [comment]
My personal memo:
```java
    InputStream in = new FileInputStream("/path/to/file.csv", "UTF-8");

    //skip UTF-8 BOM bytes:
    file.mark(3);
    if(in.read() == 239 & in.read() == 187 & in.read() == 191){
         System.out.println("UTF-8 with BOM, bytes discarded");
    } else {
         file.reset();
    }
```

**Comment by jbax** [comment]
All done, I've released a 2.5.0-SNAPSHOT build to include support for all sorts of variations around BOM markers.

Thank you for using our parsers!
